geth/peers/topicpool.go

```diff
@@ -33,24 +32,20 @@ func NewTopicPool(topic discv5.Topic, limits params.Limits, slowSync, fastSync t
 
 // TopicPool manages peers for topic.
 type TopicPool struct {
-	topic    discv5.Topic
-	limits   params.Limits
-	slowSync time.Duration
-	fastSync time.Duration
+	topic  discv5.Topic
+	limits params.Limits
 
+	mu      sync.RWMutex
+	wg      sync.WaitGroup // wait group for all goroutines spawn during TopicPool operation
```

Comment:
Hasn't there been a reason for multiple groups outside before? 🤔

Reply:
There was but I started looking into it and could find the reason. I guess it's confirmed that it's fine now: https://github.com/status-im/status-go/pull/867#discussion_r184357389